sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/DiskEncryptionSet.java

```diff
@@ -0,0 +1,236 @@
+         * access role for the key vault.
+         */
+        interface WithSystemAssignedIdentityBasedAccessOrCreate extends WithCreate {
+            /**
+             * Specifies that disk encryption set's system assigned (local) identity should have the given access
+             * (described by the role) on the current Azure key vault that's associated with it.
+             * @param builtInRole access role to assigned to the disk encryption set's local identity
+             * @return the next stage of the definition
+             */
+            WithCreate withSystemAssignedIdentityBasedAccessToCurrentKeyVault(BuiltInRole builtInRole);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to enable automatic key rotation.
+         */
+        interface WithAutomaticKeyRotation {
+            /**
+             * Enable automatic key rotation.
+             * If enabled, the system will automatically update all managed disks, snapshots, and images
+             * referencing the disk encryption set to use the new version of the key within one hour.
+             * @return the next stage of the definition
+             */
+            WithCreate withAutomaticKeyRotation();
+        }
+
+        /**
+         * The stage of the definition which contains all the minimum required inputs for the resource to be created,
+         * but also allows for any other optional settings to be specified.
+         */
+        interface WithCreate
+            extends Creatable<DiskEncryptionSet>,
+                DefinitionWithTags<WithCreate>,
+                WithAutomaticKeyRotation { }
+    }
+
+    /** The template for an update operation, containing all the settings that can be modified. */
+    interface Update
+        extends Appliable<DiskEncryptionSet>,
+        UpdateStages.WithSystemAssignedManagedServiceIdentity,
+        UpdateStages.WithSystemAssignedIdentityBasedAccess,
+        UpdateStages.WithKeyVault,
+        UpdateStages.WithAutomaticKeyRotation {
+    }
+
+    /** Grouping of disk encryption set update stages. */
+    interface UpdateStages {
+        /**
+         * The stage of a disk encryption set update allowing to enable System Assigned (Local) Managed Service
+         * Identity.
+         */
+        interface WithSystemAssignedManagedServiceIdentity {
+            /**
+             * Specifies that System Assigned (Local) Managed Service Identity needs to be enabled in the disk
+             * encryption set.
+             * @return the next stage of the update
+             */
+            WithSystemAssignedIdentityBasedAccess withSystemAssignedManagedServiceIdentity();
+
+            /**
+             * Specifies that System Assigned (Local) Managed Service Identity needs to be disabled in the disk
+             * encryption set.
+             * @return the next stage of the update
+             */
+            Update withoutSystemAssignedManagedServiceIdentity();
```

Comment:
Will this ever happen? This might just render it non-workable?

Reply:
I saw a doc about [`Transferring a subscription between Azure AD directories`](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/known-issues#transferring-a-subscription-between-azure-ad-directories). They suggest:
> Workaround for managed identities in a subscription that has been moved to another directory:
For system assigned managed identities: disable and re-enable.

Thought this might be useful in that particular situation.

Reply:
Sure. If it is useful, do keep it.